server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/SlackMessageService.java

```diff
@@ -57,26 +63,47 @@ public void sendMessage(String channelId, List<LayoutBlock> blocks, String fallb
         }
     }
 
+    private List<LeaderboardEntryDTO> getTop3SlackReviewers() {
+        LocalDate after = LocalDate.now().minusDays(8);
+        LocalDate before = LocalDate.now().minusDays(1);
+        var leaderboard = leaderboardService.createLeaderboard(Optional.of(after), Optional.of(before),
+                Optional.empty());
+        return leaderboard.subList(0, Math.min(3, leaderboard.size()));
+    }
+
     @Scheduled(cron = "${slack.cronSchedule}")
     public void sendScheduledLeaderboard() {
         if (!runScheduledMessage) {
             return;
         }
+
+        var top3reviewers = getTop3SlackReviewers();
+        logger.info("Top 3 Accounts of the last week: " + top3reviewers);
+
         logger.info("Sending scheduled message to Slack channel...");
-        var blocks = asBlocks(
+        List<LayoutBlock> blocks = asBlocks(
                 header(header -> header.text(
-                        plainText(pt -> pt.text(":newspaper:  Developer Reviews of the last week  :newspaper:")))),
+                        plainText(pt -> pt.text(":newspaper: Developer Reviews of the last week :newspaper:")))),
                 divider(),
                 context(context -> context
-                        .elements(List.of(markdownText("*October 31, 2024*  |  Hephaestus Announcement")))),
+                        .elements(List.of(markdownText("*October 31, 2024* | Hephaestus Announcement")))),
+                section(section -> section.text(markdownText(
+                        "Another *review leaderboard* has concluded. You can check out your placement at <https://hephaestus.ase.cit.tum.de/|hephaestus.ase.cit.tum.de>. The score is a weighted heuristic based on the complexities of the reviewed PRs (similar to the old algorithm)."))),
                 section(section -> section.text(markdownText(
-                        "Here is the *review table* summary for the last week. The table is taken directly from <https://hephaestus.ase.cit.tum.de/|Hephaestus> and now also includes non-review comments & code comments in addition to all variants of reviews for each developer. The score is a weighted heuristic based on the complexities of the reviewed PRs (similar to the old algorithm)."))),
+                        "If you would like to check out more leaderboards or statistics, head over to last week's leaderboard or this month's leaderboard."))),
                 section(section -> section.text(markdownText(
-                        "If you would like to check out more leaderboards or statistics, head over to last week's leaderboard or this month's leaderboard."))));
+                        "The top 3 reviewers of the last week are:"))),
+                section(section -> section.text(markdownText(
+                        IntStream.range(0, top3reviewers.size())
+                                .mapToObj(i -> ("• *" + (i + 1) + ". @" + top3reviewers.get(i).user().name() + " *"))
```

Comment:
Do you think this user matching algorithm is enough? Maybe we can be a bit smarter here? For example, my GitHub name is `Felix T.J. Dietrich` but on Slack I'm just `Felix Dietrich`

Reply:
The reviewed code was still missing the adaptation to when we are able to use the `users:read`-permission, so feel free to take look at the new version now.
Nevertheless, I don't really see another way of comparing the accounts aside from real names since there is no direct connection. We could probably do some kind of similarlity matching for the names to compensate for such differences?

Reply:
I was thinking about string edit distance maybe 🤔